wasmer/config.go

```diff
@@ -166,6 +167,36 @@ func (self *Config) UseCraneliftCompiler() *Config {
 	return self
 }
 
+var opCodeMap map[Opcode]uint32 = nil
+
+//export metering_delegate
+func metering_delegate(op C.wasmer_parser_operator_t) C.uint64_t {
+	// a simple alogorithm for now just map from opcode to cost directly
+	// all the responsibility is placed on the caller of PushMeteringMiddleware
+	v, b := opCodeMap[Opcode(op)]
+	if !b {
+		return 0 // no value means no cost
+	}
+	return C.uint64_t(v)
+}
```

Comment:
I wonder if there is a way to have this as a closure'd function so it doesn't need to modify a global variable with the opcode costs (basically to allow two different running instances to have different opcode values within the same program)

Reply:
I was trying.  The external c linkage on that function is what I was unable to figure a way around.

Reply:
But I agree what you stated would be MUCH better.

Reply:
Have you tried this?
https://stackoverflow.com/a/69637827

Reply:
I have not.  I see now.  Let me integrate this!

Reply:
@syrusakbary In looking very carefully, this solution will again not work as it requires an //export on the function that gets called back by C.  To create a closure, the closure at some point has to be called by c and that function needs to be prototypes in cgo which cannot occur dynamically.  I still don't see a way unless I am looking past the obvious or I can call some cgo runtime directly.

Reply:
Ok, let's merge the current PR as is better than the current state in master. We'll see if it's possible to do something like that in the future!